### ISSUE TITLE
[BUG] Cannot change tooltips in JS callback

### ISSUE THREAD

**Issue by tony-bony** [bokeh/bokeh#10639]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
```
Python version      :  3.8.1 (default, Dec 27 2019, 18:06:00) 
IPython version     :  7.12.0
Tornado version     :  6.0.3
Bokeh version       :  2.2.3
BokehJS static path :  /usr/local/lib/python3.8/site-packages/bokeh/server/static
node.js version     :  v13.8.0
npm version         :  6.13.7
OS: macOS Catalina 10.15.5
Google Chrome: Version 86.0.4240.111 (Official Build) (x86_64)
```

#### Description of expected behavior and the observed behavior
After clicking the button the tooltip should change from: `[('x', '@x'), ('y', '@y')]` to `[['y', '@y'], ['z', '@z']]` but nothing changes

This code works as expected for Bokeh v1.4.0 till v2.1.1 but not for the current v2.2.3

#### Complete, minimal, self-contained example code that reproduces the issue

```
from bokeh.plotting import show, figure
from bokeh.models import ColumnDataSource, CustomJS, Row, Button

source = ColumnDataSource({'x': [1, 2, 3], 'y': [1, 2, 3], 'z': [3, 2, 1], 'size': [10, 20, 30]})

p = figure(tools = 'hover', tooltips = [('x', '@x'), ('y', '@y')])
p.circle('x', 'y', size = 'size', source = source)

buttom_code = "p.toolbar.tools[0].tooltips = [['y', '@y'], ['z', '@z']]"
button = Button(label = 'Change Tooltip')
button.js_on_event('button_click', CustomJS(args = {'p': p}, code = buttom_code))

show(Row(p, button))

```

#### Stack traceback and/or browser JavaScript console output
No JS errors

**Comment by mattpap** [comment]
This is related to issue #10559, possibly a duplicate.

**Comment by mattpap** [comment]
Definitively a duplicated. Fixed in PR #10560.
